Add website content

jurraca opened the pull request:
Working branch to add content to website.

jurraca commented:
@fjahr could I ask you to handle the Contributing section ?

fjahr commented:
> @fjahr could I ask you to handle the Contributing section ?

Done, I pushed it directly into main along with a page on Bitcoin Core stuff.

We are a bit limited with the number of top level pages because the menu looks weird but also I think it's a better structure if we don't have too much there. I have been thinking a bit about this again and I would now suggest the following:

- home/index (not in menu, but could be): This has the basic information on what asmap is, the intro basically
- Research/Deep Dive: This would be the top category to all the conceptual research/information, we have sub-categories for each of the topics and I would also put the links to resources and the glossary here
  - Background
  - Data Sources
  - Impact research (i.e. the work by viritu https://github.com/virtu/talks/blob/master/2023-03-02-advancing-bitcoin/slides.pdf)
  - Resources (and Tools)
  - Glossary
- Bitcoin Core: Info on WIP, Release process stuff etc.
- Contributing

What do you think? I am not sure yet if that is the best way, resources and tools for example don't really fit in well anywhere and also it's repeating mostly what is explained and linked on the other pages as well probably, so I tend to not want to put it on the top level.

I am also not sure yet what the best way would be for this sub-category navigation, but I sure we'll be able to make it work if you like this suggestion.

jurraca commented:
yep, I think that structure makes sense. I modified it, and made the subfolder structure explicit in the index, let me know if that works. I think it flows fairly nicely. Kept the glossary at top level though. 

I also edited some of the wording of your core and contributing sections, lmk what you think. 

rebased on main with your changes
